plugins/out_kinesis_firehose/firehose.c

```diff
@@ -120,6 +120,11 @@ static int cb_firehose_init(struct flb_output_instance *ins,
         ctx->sts_endpoint = (char *) tmp;
     }
 
+    tmp = flb_output_get_property("profile", ins);
+    if (tmp) {
+        ctx->profile = (char *) tmp;
+    }
+
```

Comment:
rather than doing this everywhere, let's have all our plugins use FLB_AWS_CREDENTIAL_BASE_CONFIG_MAP now?

Reply:
Do you mean moving the base config properties in standard_chain_provider_create?

Reply:
the config map will make sure that all properties are set in the ctx which the standard_chain_provider_create can then use. this mean you do not need to call flb_output_get_property.

Reply:
As discussed, I guess the way we can use the existing standard_chain_provider_create way of doing things. So you can keep this. Just need to add the config map entry for each plugin.

Reply:
I changed my mind we don't need to use FLB_AWS_CREDENTIAL_BASE_CONFIG_MAP